lib/spack/spack/build_systems/cmake.py

```diff
@@ -213,7 +213,7 @@ def std_cmake_args(self):
         return std_cmake_args
 
     @staticmethod
-    def std_args(pkg, generator=None):
+    def std_args(pkg, generator=generator):
```

Comment:
What happens if we leave this?

Reply:
The value set [above ](https://github.com/spack/spack/blob/74466b9d2f6a169a07c1439d44872d8fa9da227a/lib/spack/spack/build_systems/cmake.py#L182) gets overridden by the None default which leads to Windows getting the default "Unix Makefiles" generator rather than "Ninja"

Reply:
I think removing the null check makes more sense. Making them positional would lead changing quite a few other places to know about what generator to use.